src/libcrun/handlers/krun.c

```diff
@@ -41,86 +41,139 @@
 /* libkrun has a hard-limit of 8 vCPUs per microVM. */
 #define LIBKRUN_MAX_VCPUS 8
 
+struct krun_config {
+  void *handle;
+  void *handle_sev;
+  bool sev;
+};
+
 /* libkrun handler.  */
 #if HAVE_DLOPEN && HAVE_LIBKRUN
 static int
 libkrun_exec (void *cookie, libcrun_container_t *container, const char *pathname, char *const argv[])
 {
   runtime_spec_schema_config_schema *def = container->container_def;
+  int32_t (*krun_set_log_level) (uint32_t level);
   int32_t (*krun_create_ctx) ();
   int (*krun_start_enter) (uint32_t ctx_id);
   int32_t (*krun_set_vm_config) (uint32_t ctx_id, uint8_t num_vcpus, uint32_t ram_mib);
   int32_t (*krun_set_root) (uint32_t ctx_id, const char *root_path);
+  int32_t (*krun_set_root_disk) (uint32_t ctx_id, const char *disk_path);
   int32_t (*krun_set_workdir) (uint32_t ctx_id, const char *workdir_path);
   int32_t (*krun_set_exec) (uint32_t ctx_id, const char *exec_path, char *const argv[], char *const envp[]);
-  void *handle = cookie;
+  int32_t (*krun_set_tee_config_file) (uint32_t ctx_id, const char *file_path);
+  struct krun_config *kconf = (struct krun_config *) cookie;
+  void *handle;
   uint32_t num_vcpus, ram_mib;
   int32_t ctx_id, ret;
   cpu_set_t set;
 
+  if (access("/krun-sev.json", F_OK) == 0)
+    {
+      if (kconf->handle_sev == NULL)
+        error (EXIT_FAILURE, 0, "the container requires libkrun-sev but it's not available");
+      handle = kconf->handle_sev;
+      kconf->sev = true;
+    }
+  else
+    {
+      if (kconf->handle == NULL)
+        error (EXIT_FAILURE, 0, "the container requires libkrun but it's not available");
```

Comment:
Nit: Not sure but Is this error message correct ? Shouldn't this be `the container requires libkrun with sev support but it's not available"` or is it that now every instance will have `krun-sev.json`

Reply:
My bad this is correct.